### PR TITLE
feat(appium): Make doctor extensible

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
## Proposed changes

Addresses https://github.com/appium/appium/issues/18302

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: packages/appium/docs/en/developing/build-doctor-checks.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+
+This tutorial is supposed to be used by plugin or driver authors that would like to help their users
+to deal with complicated setup or configuration steps.
+
+## Adding Doctor Checks
+
+### Typing Requirements
+
+The term `Doctor Check` literally describes a single javascript class instance that implements the
+[IDoctorCheck interface](../../../../types/lib/doctor.ts). The interface defines the following methods:
```
Comment (eglitise) [inline packages/appium/docs/en/developing/build-doctor-checks.md:19]: This link should be probably changed to https://github.com/appium/appium/blob/master/packages/types/lib/doctor.ts (which doesn't exist yet, but will once this PR is merged). MkDocs won't know how to present a TS file.

**Review by eglitise — COMMENTED** [review]
Would these changes eventually allow to deprecate the `@appium/doctor` package, if the checks can now be included in the extension itself?

File: packages/appium/docs/en/developing/build-doctor-checks.md (hunk 1 above)
Comment (mykola-mokhnach) [reply packages/appium/docs/en/developing/build-doctor-checks.md]: changed the link

**Comment by mykola-mokhnach** [comment]
> Would these changes eventually allow to deprecate the `@appium/doctor` package, if the checks can now be included in the extension itself?

yes, this is the idea
The next step would be to move existing checks hardcoded into `@appium/doctor` to appropriate drivers

File: packages/appium/lib/doctor/doctor.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+      log.info('');
+    };
+
+    await handleIssues([
+      '### Manual Fixes Needed ###',
+      'The configuration cannot be automatically fixed, please do the following first:',
+    ], manualIssues);
+    await handleIssues([
+      '### Optional Manual Fixes  ###',
+      'The configuration can install optionally. Please do the following manually:',
```
Comment (jlipps) [inline packages/appium/lib/doctor/doctor.js:94]:
```suggestion
      'To fix these optional configurations, do the following manually:',
```

**Review by jlipps — APPROVED** [review]
Love this!

File: packages/appium/lib/cli/plugin-command.js [hunk 3]
```diff
@@ -65,6 +65,18 @@ export default class PluginCliCommand extends ExtensionCliCommand {
     });
   }
 
+  /**
+   * Runs doctor checks for the given plugin
```
Comment (KazuCocoa) [inline packages/appium/lib/cli/plugin-command.js:69]: 👍

**Review by KazuCocoa — APPROVED** [review]
lgtm 🚀

File: packages/appium/lib/doctor/doctor.js (hunk 2 above)
Comment (mykola-mokhnach) [reply packages/appium/lib/doctor/doctor.js]: fixed

**Pull request merged**